usb: Remove default Vendor and Product ID

finikorg opened the pull request:
Settings configuration allows to set Vendor and Product IDs. These
configuration options should be used when building USB gadget
applications.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg commented:
@pfl @nashif @jfischer-phytec-iot  I do not other ways to add reviewers

jfischer-phytec-iot commented on subsys/usb/Kconfig:
```diff
@@ -16,6 +17,24 @@ menuconfig USB_DEVICE_STACK
+	Allow to set USB device stack settings including Vendor ID
+	and Product ID.
+
+if USB_DEVICE_STACK_SETTINGS
```
It makes no sense to make it optional, PID and VID must be explicitly configured.

finikorg replied:
@jfischer-phytec-iot @pfl That is the idea first to introduce those parameters to make slow transition not affecting existing build sanity checks. The problem here is that we cannot assign default values here (unless you buy vendor id, in that case feel free to send a patch).

jfischer-phytec-iot commented on subsys/usb/Kconfig:
```diff
@@ -16,6 +17,24 @@ menuconfig USB_DEVICE_STACK
+	hex "USB Device Vendor ID"
+
+config USB_DEVICE_PRODUCT_ID
+	hex "USB Device Product ID"
```
Configurable VID and PID without the ability to configure manufacturer / product strings is not really practical.

finikorg replied:
We can add later other values

jfischer-no reviewed (changes requested):
I suggest to wait for #716 or pick 566c4cd

finikorg commented:
566c4cddad26a5473dc0ae189f7eeed209011c70 is not acceptable for us, those value should not be used @jfischer-phytec-iot

nashif commented:
is this still relevant? We have those now as part of f0832da1.

finikorg commented:
@nashif This is outdated now.

finikorg committed:
usb: Remove default Vendor and Product ID

Vendor ID and Product ID should be assigned on build time by
respective process.

For sanity check we assign some random values which are only used for
build tests and should not be used for real products.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg commented:
@nashif @pfl I have updated this pull request

pfl commented:
Looks fine. If there are other hard coded values in use, cleaning up those should make it as patch number two.

pfl reviewed (approved):
Looks ok to me.

jfischer-no reviewed (changes requested):
a nice try, but it's not so simple, e.g. tinytile select usb device stack by default:
boards/x86/tinytile/tinytile_defconfig:15:CONFIG_USB_UART_CONSOLE=y

pfl commented:
Tinytile board file needs to contain proper VID and PID in order to be useful. Do we know already if and what values the board should use?

finikorg committed:
usb: Add missing tag to sanity check configuration

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg commented:
@jfischer-phytec-iot @nashif @pfl What shall we put for tinytile? This is the board which uses USB Device stack for serial console so the parameters for Vendor and Product ID should be available.

nashif reviewed (changes requested):
I am not following why are you adding prj_sanity.conf instead of changing the existing file that already exists!

finikorg commented:
@nashif prj_sanity.conf configuration files are supposed to exist only for sanity check since they put some random values for Product ID and Vendor ID.

nashif commented:
> @nashif prj_sanity.conf configuration files are supposed to exist only for sanity check since they put some random values for Product ID and Vendor ID.

Use the new extra_configs in testcase.yaml instead, this way we will avoid duplication of the base configuration, see #4374 for documentation of this new option.

finikorg commented:
@nashif this method is not working. I have:
...
      extra_configs:
        - CONFIG_USB_DEVICE_VID=0xDEAD
        - CONFIG_USB_DEVICE_PID=0xBEEF
...

And error message is:
.config:32:warning: symbol value '0xDEADCONFIG_USB_DEVICE_PID=0xBEEF' invalid for USB_DEVICE_VID

nashif committed:
sanitycheck: fix extra_configs with multiple values

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
found the bug, fixed in #4403

finikorg commented:
@nashif fixed, the issue with tinytile remains

nashif commented:
> @nashif fixed, the issue with tinytile remains

Just use whatever we have used sofar not changing the status quo here.

finikorg commented:
@nashif if we could provide those 2 lines of extra_configs for tinytile sanity_check tests that would be enough. When we get Zephyr VID we assing PID to tinytile CDC ACM.

nashif commented:
> @nashif if we could provide those 2 lines of extra_configs for tinytile sanity_check tests that would be enough. When we get Zephyr VID we assing PID to tinytile CDC ACM.

This has nothing to do with sanitycheck, this is a board configuration that needs to work outside of sanitychech

jfischer-phytec-iot commented:
tinytile has Intel inside, can we still use Intel's VID/PID for `boards/x86/tinytile/Kconfig.defconfig`?

finikorg commented:
@pfl Can we use Intel VID / PID for tinytile?

finikorg committed:
boards: tinytile: Set original firmware Vendor and Product IDs

Set Vendor and Product ID originally present in the firmware.
Implemented USB function is the same: CDC ACM serial port.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg commented:
@laperie @nashif

nashif commented:
@jfischer-phytec-iot you ok with this?

jfischer-phytec-iot commented:
@nashif LGTM
